pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -518,13 +520,16 @@ void ClientConnection::processIncomingBuffer() {
     readNextCommand();
 }
 
-bool ClientConnection::verifyChecksum(SharedBuffer& incomingBuffer_, proto::BaseCommand& incomingCmd_) {
+bool ClientConnection::verifyChecksum(SharedBuffer& incomingBuffer_, uint32_t& remainingBytes, proto::BaseCommand& incomingCmd_) {
     int readerIndex = incomingBuffer_.readerIndex();
     bool isChecksumValid = true;
+
     if (incomingBuffer_.readUnsignedShort() == Commands::magicCrc32c) {
         uint32_t storedChecksum = incomingBuffer_.readUnsignedInt();
+        remainingBytes -= (2 + 4) /* subtract size of checksum itself */;
+
         // compute metadata-payload checksum
-        int metadataPayloadSize = incomingBuffer_.readableBytes();
```

Comment:
umm.. I don't see a different here.
after reading magic number and checksum: `incomingBuffer_.readableBytes()` is having metadata+payload only.  so, essentially we are computing checksum on correct bytes only. with your change `remainingBytes=metdata+payload` only so, isn't it do the same thing?

Reply:
This was not what was failing here, but the problem is that the `incomingBuffer_` is not a slice containing only the current frame (like we have with Netty). In this case it can contain data after the current frame, so have to "manually" limit ourselves to no exceed the frame.

Reply:
I see.. then with this change if broker starts sending messages with checksum to old cpp consumer, then isn't cpp consumer will fail to validate checksum.? in this case, should broker remove checksum for consumers with version < V11 ?

Reply:
Yes, the remove is already in this PR

Reply:
> Yes, the remove is already in this PR

umm.. I don't see any related change in [consumer](https://github.com/apache/incubator-pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java#L205) file. Do i miss anything?

Reply:
you're right, I had done the change, though didn't make it to the commit